contracts/AnyswapV4CallProxy.sol

```diff
@@ -151,20 +200,25 @@ contract AnyCallProxy is Whitelistable {
         address[] memory callbacks,
         uint256[] memory nonces,
         uint256 fromChainID
-    ) external onlyMPC lock {
+    ) external onlyMPC lock bill(from) {
         require(from != address(this) && from != address(0), "AnyCall: FORBID");
+        require(!isBlacklisted[from]);
+
         uint256 length = to.length;
         bool[] memory success = new bool[](length);
         bytes[] memory results = new bytes[](length);
-        uint256 chainID = block.chainid;
+
+        context = Context({sender: from, fromChainID: fromChainID});
```

Comment:
Set the context here so integrators know who is calling on the source chainid

Reply:
This context is not logged, how do integrators use this?

Reply:
> This context is not logged, how do integrators use this?

There's a public storage value [`context`](https://github.com/anyswap/anyswap-v1-core/blob/69b4c998a56abec7fdacbb6f70e1b3e2dd13423d/contracts/AnyswapV4CallProxy.sol#L157). So integrators can have an interface like:

```solidity
interface Anyswap {
     function context() external view returns (address, uint256);
}
```

However, I can add in two view functions which get either or of the `sender` or `fromChainID` elements.

Reply:
I see. They could pass `sender`, `fromChainID` in calldata, but it's better to have this.